clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsApplicationProvider.java

```diff
@@ -0,0 +1,147 @@
+package com.netflix.spinnaker.clouddriver.ecs.view;
+
+
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.*;
+import com.google.common.collect.Sets;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsApplication;
+import com.netflix.spinnaker.clouddriver.model.Application;
+import com.netflix.spinnaker.clouddriver.model.ApplicationProvider;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class EcsApplicationProvider implements ApplicationProvider {
+
+  public static final String SERVICE_DELIMITER = "-";
+
+  AccountCredentialsProvider accountCredentialsProvider;
+  AmazonClientProvider amazonClientProvider;
+
+  @Autowired
+  public EcsApplicationProvider(AccountCredentialsProvider accountCredentialsProvider, AmazonClientProvider amazonClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.amazonClientProvider = amazonClientProvider;
+    getApplications(false);
+  }
+
+
+  @Override
+  public Application getApplication(String name) {
+
+    for (Application application: getApplications(false)) {
+      if (name.equals(application.getName())) {
+        return application;
+      }
+    }
+
+    return null;
+  }
+
+  @Override
+  public Set<Application> getApplications(boolean expand) {
+    Set<Application> applications = new HashSet<>();
+
+    for (AccountCredentials credentials: accountCredentialsProvider.getAll()) {
+      if (credentials instanceof AmazonCredentials) {
+        Set<Application> retrievedApplications = findApplicationsForAllRegions((AmazonCredentials) credentials);
```

Comment:
Why is casting to AmazonCredentials needed if the if statement is already checking that the type is AmazonCredentials?

Reply:
Because Java :)  Checking the type of the object doesn't cast the object, you still need to explicitly do it afterwards

Reply:
I was not expecting the check to cast the object. But if "credentials instanceof AmazonCredentials" == true, then (AmazonCredentials) credentials to be a no-op. 

i.e. What is the type of credentials before the cast? and if it is not AmazonCredentials, why does `credentials instacneof AmazonCredentials` evaluate to true?

Reply:
Before the cast, credentials are typed as `AccountCredentials`.  `AmazonCredentials` extends `AccountCredentials`

The method called on line 56 expects AmazonCredentials, not AccountCredentials